### PR TITLE
Implement series.apply()

### PR DESCRIPTION
**Pull request by PokhodenkoSA**

### CODE REVIEW

File: sdc/datatypes/pandas_series_functions/apply.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+@sdc_overload_method(SeriesType, 'apply')
+def hpat_pandas_series_apply(self, func):
+    """
+    """
+
+    _func_name = 'Method apply().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
```
Comment (densmirn) [inline sdc/datatypes/pandas_series_functions/apply.py]: Could you use `TypeChecker` for such cases?

File: sdc/hiframes/hiframes_typed.py [hunk 2]
```diff
@@ -985,7 +985,7 @@ def run_call_series_quantile_default(A):
         if func_name == 'combine':
             return self._handle_series_combine(assign, lhs, rhs, series_var)
 
-        if func_name in ('map', 'apply'):
+        if func_name in ('map'):
```
Comment (densmirn) [inline sdc/hiframes/hiframes_typed.py]: Looks like this change has no effect.

File: sdc/hiframes/pd_series_ext.py [hunk 3]
```diff
@@ -573,10 +573,10 @@ def _resolve_map_func(self, ary, args, kws):
...
         return self._resolve_map_func(ary, args, kws)
 
-    @bound_function("series.apply")
-    def resolve_apply(self, ary, args, kws):
-        # TODO: handle apply differences: extra args, np ufuncs etc.
-        return self._resolve_map_func(ary, args, kws)
+    # @bound_function("series.apply")
+    # def resolve_apply(self, ary, args, kws):
+    #     # TODO: handle apply differences: extra args, np ufuncs etc.
+    #     return self._resolve_map_func(ary, args, kws)
```
Comment (densmirn) [inline sdc/hiframes/pd_series_ext.py]: If you add `'resolve_apply'` to the list `_non_hpat_pipeline_attrs` in `pd_series_ext.py`, then you don't need to comment out these lines.

Comment (densmirn) [reply sdc/hiframes/pd_series_ext.py]: Shouldn't we leave `apply` in "old style" for `SDC` pipeline (`SDC_CONFIG_PIPELINESDC=1`)?

Comment (densmirn) [reply sdc/hiframes/pd_series_ext.py]: Eventually I think we can leave the lines without changes.

**Comment by densmirn** [comment]
> Hello @PokhodenkoSA! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:
> 
> * In the file [`sdc/datatypes/hpat_pandas_series_functions.py`](https://github.com/IntelPython/sdc/blob/567a09f8936a2c0f094245e87a88e4de94b28b53/sdc/datatypes/hpat_pandas_series_functions.py):
> 
> > [Line 4135:1](https://github.com/IntelPython/sdc/blob/567a09f8936a2c0f094245e87a88e4de94b28b53/sdc/datatypes/hpat_pandas_series_functions.py#L4135): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> 
> * In the file [`sdc/tests/test_series.py`](https://github.com/IntelPython/sdc/blob/567a09f8936a2c0f094245e87a88e4de94b28b53/sdc/tests/test_series.py):
> 
> > [Line 230:1](https://github.com/IntelPython/sdc/blob/567a09f8936a2c0f094245e87a88e4de94b28b53/sdc/tests/test_series.py#L230): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file

Please fix these code style issues.

File: sdc/hiframes/pd_series_ext.py (hunk 3 above)
Comment (PokhodenkoSA) [reply sdc/hiframes/pd_series_ext.py]: Done. I have used `_non_hpat_pipeline_attrs` to switch off old implementation.

File: sdc/hiframes/hiframes_typed.py (hunk 2 above)
Comment (PokhodenkoSA) [reply sdc/hiframes/hiframes_typed.py]: Reverted this changes.

File: sdc/datatypes/pandas_series_functions/apply.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    .. only:: developer
+       Test: python -m sdc.runtests sdc.tests.test_series_apply
+    """
+
+    ty_checker = TypeChecker("Method apply().")
+    ty_checker.check(self, SeriesType)
+
+    def impl(self, func, convert_dtype=True, args=()):
+        return pandas.Series(list(map(func, self._data)), self.index)
```
Comment (AlexanderKalistratov) [inline sdc/datatypes/pandas_series_functions/apply.py]: What about series name?

Comment (AlexanderKalistratov) [reply sdc/datatypes/pandas_series_functions/apply.py]: Shouldn't it also pass `args`?

Comment (PokhodenkoSA) [reply sdc/datatypes/pandas_series_functions/apply.py]: Does not work with `args`. `func` should be replaced with `lambda x: func(x, *args)`. It does not work for me.

Comment (PokhodenkoSA) [reply sdc/datatypes/pandas_series_functions/apply.py]: Should I generate function on the fly and unroll `args`?

Comment (PokhodenkoSA) [reply sdc/datatypes/pandas_series_functions/apply.py]: > What about series name?

Done

File: sdc/tests/test_series_apply.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    def test_series_apply_lambda(self):
+        def test_impl(S):
+            return S.apply(lambda a: 2 * a)
+        hpat_func = self.jit(test_impl)
+
+        S = pd.Series(DATA)
+        pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
+
+    @skip_sdc_jit("'Var' object has no attribute 'py_func'")
```
Comment (AlexanderKalistratov) [inline sdc/tests/test_series_apply.py]: Please also add test for `args` case and skip it

Comment (PokhodenkoSA) [reply sdc/tests/test_series_apply.py]: Done

**Pull request merged**